documentation/API-design-guidelines.md

```diff
@@ -1976,40 +1976,46 @@ response:
 204 No Content
 ```
 
-## Appendix A: `info.description` template for `device` identification from access token
+## Appendix A (Normative): `info.description` template for when User identification can be from either an access token or explicit identifier
 
-The documentation template below is recommended to be used as part of the API documentation in `info.description` property in the CAMARA API specs which use the `device`object defined in [CAMARA_common.yaml](/artifacts/CAMARA_common.yaml) artifact. This template provides guidance on how to handle device information in API requests **when using 3-legged access tokens and the device can be uniquely identified by the token**.
+When an API requires a User (as defined by the [ICM Glossary](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md#glossary-of-terms-and-concepts)) to be identified in order to get access to that User's data (as Resource Owner), the User can be identified in one of two ways:
+- If the access token is a Three-Legged Access Token, then the User will already have been associated with that token by the API provider, which in turn may be identified from the physical device that calls the `/authorize` endpoint for the OIDC authorisation code flow, or from the `login_hint` parameter of the OIDC CIBA flow (which can be a device IP, phone number or operator token). The `sub` claim of the ID token returned with the access token will confirm that an association with the User has been made, although this will not identify the User directly.
```

Comment:
Can you clarify your concern?

I understood it as a "will", because the [Security and Interoperability](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-Security-Interoperability.md#id-token-sub-claim) profile states:
> The sub claim MUST not contain Personally Identifiable Information (PII) such as the MSISDN.

So the `sub` itself will not directly allow the User to be identified, as it cannot contain PII.

The point I am trying to convey here is that the existence of the `sub` confirms that a User (for consent purposes) has already been identified by the API provider, but the value of the `sub` itself does not directly identify that user.

The API consumer may anyway know the User, of course, depending upon how the 3-legged token was obtained, but they will not know just from the `sub`. Rather, that will be because of additional information they have.

Reply:
The point is that "sub" may be used to identify the user behind, and it does not mean that "a Personally Identifiable Information (PII) such as the MSISDN." has to be used. A Telco operator can use "opaque" identifiers as "sub" values. So the comment is not in such a way.

It is in the way a "User" may have, for instance, several phone numbers and decides to use the "sub" as the way to identify the User itself and additional information to identify/associate the phone number. Just your sentence in the inverse way:

> The API consumer may anyway know the User, of course, depending upon how the 3-legged token was obtained, but they will not know just from the sub. Rather, that will be because of additional information they have.

"sub" may identify the user and additional info may be used to identify/associate user device.

At the end it is an implementation topic, because Access Token will be something opaque to the API Consumer and the Telco Operator will own the needed information to manage that token suitably.

That is the reason to modify "will" by "might". 

Relevant message is this: 
 - "The `sub` claim of the ID token returned with the access token will confirm that an association with the User has been made" 
The final part is implementation dependant

Reply:
@PedroDiez
I still don't get your concern. If you say that the `sub` **"might not"** directly identify the User to the client, then you are also saying that it **"might"** directly identify the User to the client.

"directly" in this context means "within the value of the `sub` itself".

The `sub` **NEVER** directly identifies the User to the client. There is no circumstance where knowing only the `sub` (without any additional context) allows anyone to directly identify the User, and this property of the `sub` does not depend upon how it is generated by the API provider (if they are following the rules).

So the API consumer can inspect their ID token and see that a `sub` has been defined (let's say `sub: 88e399ee-ba9f-4fef-9445-016b03659991`), but they can never know the identity of that User directly from the `sub`. They might <i>think</i> they know the identity from previous transactions with the authorisation server, and they might be right, but that is **indirectly** identifying the User.

So the client may be able to indirectly identify the User from the `sub`, but never directly.

Reply:
I say it would be possible to identify a User by means of "sub". The sentence indicates will not, so forbids that option. I am not ok with that.

I undertsand your view is different "The sub NEVER directly identifies the User to the client".

In OIDC indicates the "sub" identifies the End-User https://openid.net/specs/openid-connect-core-1_0.html#SubjectIDTypes. A Telco Operator can decide which is their underlying model. 


To me is fine just to have "The `sub` claim of the ID token returned with the access token will confirm that an association with the User has been made". This is what is relevant to API Consumer to understand the situation

Reply:
@eric-murray 
Get the point, I did not refer to the use of a PII identifier, as commented durign the meeting. 

To avoid any misunderstanding, think we could add within the paragraph **bold text below** (think is reasonable and aligned with what the rationale mentioned in this thread).

- If the access token is a Three-Legged Access Token, then the User will already have been associated with that token by the API provider, which in turn may be identified from the physical device that calls the `/authorize` endpoint for the OIDC authorisation code flow, or from the `login_hint` parameter of the OIDC CIBA flow (which can be a device IP, phone number or operator token). The `sub` claim of the ID token returned with the access token will confirm that an association with the User has been made, although this will not identify the User directly **given the `sub` will not be a globally unique identifier nor contain PII as per [CAMARA ICM Security and Interoperability](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-Security-Interoperability.md#id-token-sub-claim).**

Reply:
WDYT?

Reply:
@PedroDiez
Yes, I'm happy with that clarification. I'll update the PR accordingly.

Reply:
OK, done

@rartych
I'm using references to the main branch of ICM for now. We will need to remember to update these when the next release number of the ICM documentation is known.

Reply:
Thanks for considering @eric-murray.
Given the OK